### ISSUE TITLE
Sorbet fails to run on Mac OS 10.12

### ISSUE THREAD

**Issue by andyw8** [sorbet/sorbet#946]
I'm following the steps on https://sorbet.org/docs/adopting

```
% bundle exec srb
No sorbet/ directory found. Maybe you want to run 'srb init'?

A type checker for Ruby
```

```
% bundle exec srb typecheck -e 'puts "Hello, world!"'
dyld: Symbol not found: __ZTISt18bad_variant_access
  Referenced from: /Users/awaite-dev/.rvm/gems/ruby-2.6.3/gems/sorbet-static-0.4.4263-universal-darwin-16/libexec/sorbet
  Expected in: /usr/lib/libc++.1.dylib
 in /Users/awaite-dev/.rvm/gems/ruby-2.6.3/gems/sorbet-static-0.4.4263-universal-darwin-16/libexec/sorbet
/Users/awaite-dev/.rvm/gems/ruby-2.6.3/gems/sorbet-0.4.4263/bin/srb: line 43: 32448 Abort trap: 6           "${sorbet[0]}" "${args[@]}"
```

**Comment by azdavis** [comment]
Could you let us know what version of macOS you're running? Try `sw_vers` at the command line.

**Comment by andyw8** [comment]
```
% sw_vers
ProductName:    Mac OS X
ProductVersion: 10.12.6
BuildVersion:   16G2016
```

**Comment by jonasoberschweiber** [comment]
I get the same message on the same version of macOS.

**Comment by jasnow** [comment]
I got this error message too.
Ran "sw_vers" and got:
```
ProductName:	Mac OS X
ProductVersion:	10.12.6
BuildVersion:	16G2016
```
Found this by Googling: https://forums.factorio.com/65145

**Comment by jasnow** [comment]
@andyw8 - Got around "Not Found" message by removing ***, :group => :development** part in Gemfile.

**Comment by davidpatters0n** [comment]
@jasnow - Removing `group => :development` doesn't help in my case. Still produces the same output for me: 

```
david.patterson@CATORDPA1-ML ~/C/boxoffice (sorbet) [134]> sw_vers
ProductName:	Mac OS X
ProductVersion:	10.12.6
BuildVersion:	16G2016
david.patterson@CATORDPA1-ML ~/C/boxoffice (sorbet)> bundle exec srb typecheck -e 'puts "Hello, world!"'
dyld: Symbol not found: __ZTISt18bad_variant_access
  Referenced from: /Users/david.patterson/Code/boxoffice/.bundle/ruby/2.3.0/gems/sorbet-static-0.4.4270-universal-darwin-16/libexec/sorbet
  Expected in: /usr/lib/libc++.1.dylib
 in /Users/david.patterson/Code/boxoffice/.bundle/ruby/2.3.0/gems/sorbet-static-0.4.4270-universal-darwin-16/libexec/sorbet
/Users/david.patterson/Code/boxoffice/.bundle/ruby/2.3.0/gems/sorbet-0.4.4270/bin/srb: line 43: 99522 Abort trap: 6           "${sorbet[0]}" "${args[@]}"
david.patterson@CATORDPA1-ML ~/C/boxoffice (sorbet) [134]>
```

**Comment by jasnow** [comment]
@davidpatters0n - Agree - Removing ... doesn't fix __ZTISt18bad_variant_access error.

**Comment by ptarjan** [comment]
@DarkDimius maybe this has to do with us compiling on one version of Mac and then trying to run on another? Is there something we should be doing to help with this cross compilation?

**Comment by jasnow** [comment]
Here is the ***git diff*** today after I ran "bundle update:
```
-    sorbet (0.4.4271)
-      sorbet-static (= 0.4.4271)
-    sorbet-runtime (0.4.4271)
-    sorbet-static (0.4.4271-universal-darwin-14)
+    sorbet (0.4.4289)
+      sorbet-static (= 0.4.4289)
+    sorbet-rails (0.1.3)
+    sorbet-runtime (0.4.4289)
+    sorbet-static (0.4.4289-universal-darwin-16)
```
I can capture the "srb init" output if needed.

**Comment by DarkDimius** [comment]
We're still working on this issue. We'll let you know when it's fixed(by closing this issue)

**Comment by jasnow** [comment]
FYI: Here is gist of what happens when I run "srb init" on Ruby 2.6.3 and Rails 6.0.0.rc1 repo: https://gist.github.com/jasnow/9b2b19dea6520d8a68693ee87ee4a6b7

**Comment by jasnow** [comment]
@elliott - Verified this fix on my Mac OS 10.12.6 machine. - THANK YOU!
